src/ucs/datastruct/callbackq.h

```diff
@@ -192,6 +193,33 @@ void ucs_callbackq_remove_if(ucs_callbackq_t *cbq, ucs_callbackq_predicate_t pre
                              void *arg);
 
 
+/**
+ * Add a slowpath oneshot callback to the queue.
+ * This can be used from any context and any thread.
+ *
+ * @param  [in] cbq      Callback queue to add the callback to.
+ * @param  [in] key      User-defind key, used to remove the callback later.
+ * @param  [in] cb       Callback to add.
+ * @param  [in] arg      User-defined argument for the callback.
+ */
+void ucs_callbackq_add_oneshot(ucs_callbackq_t *cbq, ucs_callbackq_key_t key,
+                               ucs_callback_t cb, void *arg);
+
+
+/**
+ * Remove all slowpath callbacks from the queue with the given key and for which
+ * the given predicate returns "true" (nonzero) value.
+ * This can be used from any context and any thread, except from the callback.
```

Comment:
What will be the behavior in case of a call from a callback?

Reply:
removed this comment